ldap/servers/slapd/mapping_tree.c

```diff
@@ -918,6 +918,14 @@ mapping_tree_node_build_tree()
      */
     int ent_count = 0;
     slapi_pblock_get(pb, SLAPI_NENTRIES, (void *)&ent_count);
+    if (ent_count == 0) {
```

Comment:
Just for curiosity, at the point "_Slapi_Entry **entries_" is not NULL, in which case ent_count could it be 0 ?

Reply:
@tbordaz Good catch, looking at search_internal_pb then entries is always set to entries size + 1, but ent_count could be 0 so I think I need to do a change like:

```
diff --git a/ldap/servers/slapd/mapping_tree.c b/ldap/servers/slapd/mapping_tree.c
index 2cc4bf82c..6df27163f 100644
--- a/ldap/servers/slapd/mapping_tree.c
+++ b/ldap/servers/slapd/mapping_tree.c
@@ -936,8 +936,8 @@ mapping_tree_node_build_tree()
     }
     slapi_ch_free((void **)&ordered_suffixes);

-    slapi_free_search_results_internal(pb);
 build_tree_done:
+    slapi_free_search_results_internal(pb);
     slapi_pblock_destroy(pb);
     slapi_ch_free((void **)&filter);
     return result;
```

Reply:
See #4502 @tbordaz